shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/core/api/impl/AbstractPipelineJobAPIImpl.java

```diff
@@ -85,23 +87,31 @@ private String convertJobConfigurationToText(final PipelineJobConfiguration jobC
     @Override
     public void startDisabledJob(final String jobId) {
         log.info("Start disabled pipeline job {}", jobId);
+        MigrationDistributedCountDownLatch.getInstance().removeParentNode(PipelineMetaDataNode.getScalingJobBarrierDisablePath(jobId));
         JobConfigurationPOJO jobConfigPOJO = getElasticJobConfigPOJO(jobId);
         if (!jobConfigPOJO.isDisabled()) {
             throw new PipelineVerifyFailedException("Job is already started.");
         }
         jobConfigPOJO.setDisabled(false);
         jobConfigPOJO.getProps().remove("stop_time");
         PipelineAPIFactory.getJobConfigurationAPI().updateJobConfiguration(jobConfigPOJO);
+        String barrierPath = PipelineMetaDataNode.getScalingJobBarrierEnablePath(jobId);
+        MigrationDistributedCountDownLatch.getInstance().register(barrierPath, jobConfigPOJO.getShardingTotalCount());
+        MigrationDistributedCountDownLatch.getInstance().await(barrierPath, 10, TimeUnit.SECONDS);
```

Comment:
Could we wait on a shorter time? e.g. 3 seconds or 5 seconds?

Reply:
Not neet to clean up when timeout, the cleanup method at `removeParentNode` will be execute at the opposite method